comp_loinc/build.py

```diff
@@ -1,83 +1,163 @@
-import typer
-from typing import List
-from ingest.part_ingest import PartOntology
-from ingest.code_ingest import CodeIngest
+"""CLI
+
+Resources
+- CLI type system docs (typer is a higher leve wrapper around click): https://click.palletsprojects.com/en/8.1.x/api/
+
+TODO's
+  1. Named args: Change args from positional to named, e.g.: https://typer.tiangolo.com/tutorial/parameter-types/number/
+"""
 import os
 import subprocess
+from pathlib import Path
+
+import typer
+
+try:
+    from comp_loinc.ingest.part_ingest import PartOntology
+    from comp_loinc.ingest.code_ingest import CodeIngest
+except ModuleNotFoundError:
+    from ingest.part_ingest import PartOntology
+    from ingest.code_ingest import CodeIngest
+
+
+app = typer.Typer(help='CompLOINC. A tool for creating an OWL version of LOINC.')
+PROJECT_DIR = Path(os.path.dirname(__file__)).parent
+DEFAULTS = {
+    'schema_file.parts': os.path.join(PROJECT_DIR, 'model/schema/part_schema.yaml'),
+    'schema_file.codes': os.path.join(PROJECT_DIR, 'model/schema/code_schema.yaml'),
+    'schema_file.composed': os.path.join(PROJECT_DIR, 'model/schema/grouping_classes_schema.yaml'),
+    'part_directory': os.path.join(PROJECT_DIR, 'data', 'part_files'),
+    'code_file': os.path.join(PROJECT_DIR, 'model/schema/code_schema.yaml'),
+    'composed_classes_data_file': os.path.join(PROJECT_DIR, 'data/composed_classes_data.yaml'),
+    'owl_directory': os.path.join(PROJECT_DIR, 'data/output/'),
+    'merged_owl': os.path.join(PROJECT_DIR, 'merged_loinc.owl'),
+    'owl_reasoner': os.path.join(PROJECT_DIR, 'elk'),
+    'output.parts': os.path.join(PROJECT_DIR, 'data/output/part_classes.owl'),
+    'output.codes': os.path.join(PROJECT_DIR, 'data/output/code_classes_cli.owl'),
+    'output.composed': os.path.join(PROJECT_DIR, 'data/output/composed_component_classes.owl'),
+    'output.merge': os.path.join(PROJECT_DIR, 'data/output/merged_loinc.owl'),
+    'output.reason': os.path.join(PROJECT_DIR, 'merged_reasponed_loinc.owl'),
+}
 
-app = typer.Typer()
 
 @app.command(name='parts')
 def build_part_ontology(
-        schema_file: str,
-        part_directory: str,
-        output: str):
-    """
-    TODO: REfactor this to use the new simplified part_ingest.py
-    :param schema_file:
-    :param part_directory:
-    :param output:
-    :return:
+    schema_file: Path = typer.Option(default=DEFAULTS['schema_file.parts'], resolve_path=True, exists=False),
```

Comment:
About: `typer.Option(..., exists=False)`

For all of the path parameters, I added this `exists=False`. I wanted it to be `exists=True`, which basically will return an error if there is no file/directory at that path. However, I believe that `typer` has a bug.

The bug is that it does not handle relative pathing correctly. It _should_ understand that relative paths should be resolved to the `pwd` where the command is being run from. But instead, it tries to resolve them relative to somewhere else.

I went to open up a new `typer` issue, but it is not at all well maintained. Just look at what it asks for if I want to open an issue; it suggests that I contribute to 2 other issues first: https://github.com/tiangolo/typer/issues/new?assignees=&labels=question&template=question.yml

So I'm leaving these as `exists=False` as a reminder that we might want to add a way to gracefully exit the program if a file path does not resolve. I also added a TODO at the top of the file.